experiments/tf_trainer/common/model_trainer.py

```diff
@@ -287,16 +320,42 @@ def export(self, serving_input_fn, example_key_name=None):
       example includes an example_key field that is passed along by the estimator
       and returned in the predictions.
     """
+    if not metrics_key and FLAGS.n_export == -1:
+      raise ValueError(
+        'Must provide value for `metrics_key` when exporting best checkpoint.')
+
     estimator = self._estimator
     if example_key_name:
       estimator = self._add_estimator_key(self._estimator, example_key_name)
-    
+
     checkpoints_to_export = self._get_list_checkpoint(FLAGS.n_export,
```

Comment:
Can we insert all this logic in the function _get_list_checkpoint? I think this maps the logic of previous function

Reply:
Done.